konfera/models/schedule.py

```diff
@@ -23,3 +25,35 @@ class Schedule(KonferaModel):
 
     def __str__(self):
         return _('%(start)s (%(duration)s min)') % {'start': self.start, 'duration': self.duration}
+
+    def clean(self, *args, **kwargs):
+        # Only approved talk can be scheduled
+        if self.talk.status != 'approved':
+            raise ValidationError(_('You cannot schedule unapproved talks'))
+
+        # Event is related to location and location has room,
+        # make sure selected room in schedule belongs to Event's location
+        room = get_model('Room').objects.filter(location=event.location)
+        if location_room.first() != self.room:
+            raise ValidationError(_('The room that you have chosen is '
+                'not part of the scheduled room'))
+
+        # Make sure date and time is within the event's range
+
+        # Make sure system does not allow store two events at the same
+        # time in the same room, eg. schedule datetime + duration in room is unique.
+        schedules = get_model('Schedule').objects.filter(
+            start=self.start, duration=self.duration, room=self.room)
+        if schedules.exists():
+            # this means that there are events with the same schedule
+            raise ValidationError(_('Another event has the same schedule.'))
+
+        return super(Schedule, self).clean(*args, **kwargs)
+
+    def save(self, *args, **kwargs):
+        self.full_clean()
+        if not self.id:
+            # If talk is selected duration is copied from talk
+            self.duration = self.talk.duration
```

Comment:
The comment says something else than the code actually does.

What is really going on in the code is:
If the schedule is being created  set the duration to the duration of the talk. (if no talk is selected it raises an error)

So `if self.talk` is the condition you need to write.